src/main/java/org/opensearch/jobscheduler/rest/RestGetJobTypeAction.java

```diff
@@ -0,0 +1,131 @@
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.rest.BytesRestResponse;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+/**
+ * This class consists of the REST handler to GET job type from extensions.
+ */
+public class RestGetJobTypeAction extends BaseRestHandler {
+
+    public static final String GET_JOB_TYPE_ACTION = "get_job_type_action";
+
+    private final Logger logger = LogManager.getLogger(RestGetJobTypeAction.class);
+
+    public JobDetailsService jobDetailsService;
+
+    @Override
+    public String getName() {
+        return GET_JOB_TYPE_ACTION;
+    }
+
+    public RestGetJobTypeAction(final JobDetailsService jobDetailsService) {
+        this.jobDetailsService = jobDetailsService;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return unmodifiableList(
+            asList(new Route(PUT, String.format(Locale.ROOT, "%s/%s", JobSchedulerPlugin.JS_BASE_URI, "_get/_job_type")))
+        );
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+        XContentParser parser = restRequest.contentParser();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+        GetJobTypeRequest getJobTypeRequest = GetJobTypeRequest.parse(parser);
+
+        final JobDetails[] jobDetailsResponseHolder = new JobDetails[1];
+
+        String jobType = getJobTypeRequest.getJobType();
+        String extensionId = getJobTypeRequest.getExtensionId();
+
+        CompletableFuture<JobDetails[]> inProgressFuture = new CompletableFuture<>();
+
+        jobDetailsService.processJobDetailsForExtensionId(
+            null,
+            jobType,
+            null,
+            null,
+            extensionId,
+            JobDetailsService.JobDetailsRequestType.JOB_TYPE,
+            new ActionListener<>() {
+                @Override
+                public void onResponse(JobDetails jobDetails) {
+                    jobDetailsResponseHolder[0] = jobDetails;
+                    inProgressFuture.complete(jobDetailsResponseHolder);
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    logger.info("could not process job type", e);
+                    jobDetailsResponseHolder[0] = null;
+                    inProgressFuture.complete(jobDetailsResponseHolder);
+                }
+            }
+        );
+
+        try {
+            inProgressFuture.orTimeout(JobDetailsService.TIME_OUT_FOR_REQUEST, TimeUnit.SECONDS);
+            inProgressFuture.get();
```

Comment:
This is the correct sequential way to handle (or you could chain `.get()` on the previous line, or `.join()`.  Same comments regarding handling timeout exception differently than an exception during execution.

Reply:
done